File: functions/src/models/UserRank.ts
export class UserRank {

    static HOUSE_RANK = "houseRank"
    static SEMESTER_RANK = "semesterRank"

    houseRank: number
    semesterRank: number

    constructor(houseRank: number, semesterRank: number){
        this.houseRank = houseRank
        this.semesterRank = semesterRank
    }

    toJson(){
        const data = {}
        data[UserRank.HOUSE_RANK] = this.houseRank
        data[UserRank.SEMESTER_RANK] = this.semesterRank
        return data
    }

}

File: lib/Models/UserRank.dart


class UserRank{

  static const String API_HOUSE_RANK = "houseRank";
  static const String API_SEMESTER_RANK = "semesterRank";


  num houseRank = -1;
  num semesterRank = -1;

  UserRank({this.houseRank, this.semesterRank});

  factory UserRank.fromJson(Map<String, dynamic> json){
    return UserRank(
        houseRank: json[API_HOUSE_RANK],
        semesterRank : json[API_SEMESTER_RANK]
    );
  }
}